src/scripts/achievements/AchievementType.ts

```diff
@@ -1,17 +1,33 @@
-enum AchievementType {
-    'money' = 0,
-    'dungeonToken' = 1,
-    'attack' = 2,
-    'diamond' = 3,
-    'itemDiggedUp' = 4,
-    'dig' = 5,
-    'capture' = 6,
-    'defeat' = 7,
-    'shiny' = 8,
-    'egg' = 9,
-    'ball' = 10,
-    'route' = 11,
-    'gym' = 12,
-    'dungeon' = 13,
-    'click' = 14
-}
\ No newline at end of file
+///<reference path="MoneyRequirement.ts"/>
+///<reference path="TokenRequirement.ts"/>
+///<reference path="AttackRequirement.ts"/>
+///<reference path="DiamondRequirement.ts"/>
+///<reference path="UndergroundItemsFoundRequirement.ts"/>
+///<reference path="UndergroundLayersMinedRequirement.ts"/>
+///<reference path="CapturedRequirement.ts"/>
+///<reference path="DefeatedRequirement.ts"/>
+///<reference path="ShinyPokemonRequirement.ts"/>
+///<reference path="HatchRequirement.ts"/>
+///<reference path="PokeballRequirement.ts"/>
+///<reference path="RouteKillRequirement.ts"/>
+///<reference path="ClearGymRequirement.ts"/>
+///<reference path="ClearDungeonRequirement.ts"/>
+///<reference path="ClickRequirement.ts"/>
+
+const AchievementType = {
+    'money': MoneyRequirement.name,
+    'dungeonToken': TokenRequirement.name,
+    'attack': AttackRequirement.name,
+    'diamond': DiamondRequirement.name,
+    'itemDiggedUp': UndergroundItemsFoundRequirement.name,
+    'dig': UndergroundLayersMinedRequirement.name,
+    'capture': CapturedRequirement.name,
+    'defeat': DefeatedRequirement.name,
+    'shiny': ShinyPokemonRequirement.name,
+    'egg': HatchRequirement.name,
+    'ball': PokeballRequirement.name,
+    'route': RouteKillRequirement.name,
+    'gym': ClearGymRequirement.name,
+    'dungeon': ClearDungeonRequirement.name,
+    'click': ClickRequirement.name,
+};
```

Comment:
i dont know if this is best, but i don't know a way to get it automatically from files, so I just change to names of classes and use this to filter